secretstores/vault/vault.go

```diff
@@ -0,0 +1,276 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package vault
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/net/http2"
+
+	"github.com/dapr/components-contrib/secretstores"
+)
+
+const (
+	defaultVaultAddress          string = "https://127.0.0.1:8200"
+	componentVaultAddress        string = "vaultAddr"
+	componentCaCert              string = "caCert"
+	componentCaPath              string = "caPath"
+	componentCaPem               string = "caPem"
+	componentSkipVerify          string = "skipVerify"
+	componentTLSServerName       string = "tlsServerName"
+	componentVaultTokenMountPath string = "vaultTokenMountPath"
+	componentVaultKVPrefix       string = "vaultKVPrefix"
+	defaultVaultKVPrefix         string = "dapr"
+)
+
+// vaultSecretStore is a secret store implementation for HashiCorp Vault
+type vaultSecretStore struct {
+	client              *http.Client
+	vaultAddress        string
+	vaultTokenMountPath string
+	vaultKVPrefix       string
+}
+
+// tlsConfig is TLS configuration to interact with HashiCorp Vault
+type tlsConfig struct {
+	vaultCAPem      string
+	vaultCACert     string
+	vaultCAPath     string
+	vaultSkipVerify bool
+	vaultServerName string
+}
+
+// NewHashiCorpVaultSecretStore returns a new HashiCorp Vault secret store
+func NewHashiCorpVaultSecretStore() secretstores.SecretStore {
+	return &vaultSecretStore{
+		client: &http.Client{},
+	}
+}
+
+// Init creates a HashiCorp Vault client
+func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
+	props := metadata.Properties
+
+	// Get Vault address
+	address := props[componentVaultAddress]
+	if address == "" {
```

Comment:
Can we make this a required property and add an error if not found or empty?

Reply:
@yaron2 we default to the local address when not found.

```
if address == "" {
	v.vaultAddress = defaultVaultAddress
}
```

Reply:
Ok